### PR TITLE
feat(advanced search): Add component to show all advanced search filters & add new filter

### PR DESCRIPTION
**Pull request by gabe-lyons**
This pr adds the root component for advanced search, including add a new filter & choose union type (AND vs OR)

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/search/AdvancedSearchAddFilterSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                value: 'value',
+                label: (
+                    <div>
+                        <StyledPlus />
+                        Add Filter
+                    </div>
+                ),
+            }}
+            labelInValue
+            style={{ padding: 6, fontWeight: 500 }}
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/search/AdvancedSearchAddFilterSelect.tsx:33]: I know we're trying to use styled components across our React app, can we do that here instead?

File: datahub-web-react/src/app/search/AdvancedSearchAddFilterSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            onChange={onFilterFieldSelect}
+            dropdownMatchSelectWidth={false}
+            filterOption={(_, option) => option?.value === 'null'}
+        >
+            {Object.keys(FIELD_TO_LABEL)
+                .sort((a, b) => FIELD_TO_LABEL[a].localeCompare(FIELD_TO_LABEL[b]))
+                .map((key) => (
+                    <Option
+                        disabled={key === 'entity' && !!selectedFilters.find((filter) => filter.field === 'entity')}
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/search/AdvancedSearchAddFilterSelect.tsx:42]: what is this function doing? It's a bit hard to tell what's going on

File: datahub-web-react/src/app/search/AdvancedSearchFilterOverallUnionTypeSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    }
+`;
+
+export const AdvancedSearchFilterOverallUnionTypeSelect = ({ unionType, onUpdate }: Props) => {
+    return (
+        <>
+            <StyledSelect
+                showArrow={false}
+                bordered={false}
+                value={unionType === UnionType.AND ? 'all filters' : 'any filter'}
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/search/AdvancedSearchFilterOverallUnionTypeSelect.tsx:29]: can we make these strings consts?

File: datahub-web-react/src/app/search/AdvancedSearchFilterOverallUnionTypeSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    onUpdate: (newValue: UnionType) => void;
+};
+
+const { Option } = Select;
+
+const StyledSelect = styled(Select)`
+    border-radius: 5px;
+    background: ${ANTD_GRAY[4]};
+    :hover {
+        background: ${ANTD_GRAY[4.5]};
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/search/AdvancedSearchFilterOverallUnionTypeSelect.tsx:19]: does this 4 vs 4.5 make a substantial visual difference?

File: datahub-web-react/src/app/search/AdvancedSearchFilters.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        setFilterField(value.value);
+    };
+
+    const onSelectValueFromModal = (values) => {
+        if (!filterField) return;
+
+        const newFilter: FacetFilterInput = {
+            field: filterField,
+            values: values as string[],
+            value: '', // TODO(Gabe): remove once we refactor the model
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/search/AdvancedSearchFilters.tsx:71]: is this a todo for after this pr?

File: datahub-web-react/src/app/search/AdvancedSearchFilters.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+            condition:
+                FIELDS_THAT_USE_CONTAINS_OPERATOR.indexOf(filterField) > -1
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/search/AdvancedSearchFilters.tsx:73]: can we use `.includes` again?

File: datahub-web-react/src/app/search/AdvancedSearchFilters.tsx (hunk 5 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/search/AdvancedSearchFilters.tsx:71]: yep

File: datahub-web-react/src/app/search/AdvancedSearchFilterOverallUnionTypeSelect.tsx (hunk 4 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/search/AdvancedSearchFilterOverallUnionTypeSelect.tsx:19]: yes!

File: datahub-web-react/src/app/search/AdvancedSearchAddFilterSelect.tsx (hunk 1 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/search/AdvancedSearchAddFilterSelect.tsx:33]: ach I wanted to do that here too but the antd `Select` was overwriting them

File: datahub-web-react/src/app/search/AdvancedSearchFilterOverallUnionTypeSelect.tsx (hunk 3 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/search/AdvancedSearchFilterOverallUnionTypeSelect.tsx:29]: this is just for display purposes, so i dont think thats needed (the actual constant is UnionType). Addinga comment

File: datahub-web-react/src/app/search/AdvancedSearchFilters.tsx (hunk 6 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/search/AdvancedSearchFilters.tsx:73]: nice catch

**Review by aditya-radhakrishnan — APPROVED** [review]

**Pull request merged**